src/a11y-dialog.ts

```diff
@@ -193,11 +193,13 @@ export default class A11yDialog {
     }
 
     // If the dialog is shown and the ESC key is pressed, prevent any further
-    // effects from the ESC key and hide the dialog, unless its role is
-    // `alertdialog`, which should be modal
+    // effects from the ESC key and hide the dialog, unless:
+    // - its role is `alertdialog`, which means it should be modal
+    // - or it contains an open popover, in which case ESC should close it
     if (
       event.key === 'Escape' &&
-      this.$el.getAttribute('role') !== 'alertdialog'
+      this.$el.getAttribute('role') !== 'alertdialog' &&
+      !this.$el.querySelector('[popover]:not([popover="manual"]):popover-open')
```

Comment:
What's this exception for?

also, as a headsup, more popover types besides manual and auto could be introduced in the future… there aren't specific discussions yet so may be a while.

Reply:
This prevents pressing `ESC` from closing the dialog when it contains an open (non-manual) popover. Because the `ESC` key was to close the popover, not the dialog.

Reply:
This selector seems to be incompatible with Firefox, see: https://github.com/discourse/discourse/actions/runs/6385717276/job/17331031221?pr=23747

```
Firefox 118.0 - [undefined ms] - Global error: SyntaxError: Element.querySelector: '[popover]:not([popover="manual"]):popover-open' is not a valid selector (…)
```

Reply:
Damn… I’ll issue a fix.

Reply:
I cannot reproduce it on Firefox 117. Odd.

Reply:
This should be fixed in v8.0.4. :)